Darwin/StitchCore/StitchCoreTests/Core/StitchCustomUserDataIntTests.swift

```diff
@@ -0,0 +1,159 @@
+        _ = try self.addRule(
+            toService: svc.1,
+            withConfig: RuleCreator.mongoDb(
+                database: dbName,
+                collection: collName,
+                roles: [RuleCreator.Role(
+                    read: true, write: true
+                    )],
+                schema: RuleCreator.Schema(properties: Document()))
+        )
+
+        _ = try self.addProvider(toApp: self.app,
+                                 withConfig: .userpass(
+                                    emailConfirmationURL: "http://emailConfirmUrl.com",
+                                    resetPasswordURL: "http://resetPasswordUrl.com",
+                                    confirmEmailSubject: "email subject",
+                                    resetPasswordSubject: "password subject"))
+
+        self.client = try self.appClient(forApp: app.0)
+    }
+
+    func testCustomUserData() throws {
+        try prepareService()
+        try self.app.customUserData.create(
+            data: CustomUserConfigData(mongoServiceId: svcId,
+                                       databaseName: dbName,
+                                       collectionName: collName,
+                                       userIdField: "recoome",
+                                       enabled: true), empty: true)
+        _ = try self.app.functions.create(data: FunctionCreator.init(
+            name: "addUserProfile",
+            source: """
+            exports = async function(color) {
+            const coll = context.services.get("mongodb1")
+            .db("\(dbName)").collection("\(collName)");
+            await coll.insertOne({
+            "recoome": context.user.id,
+            "favoriteColor": "blue"
+            });
+            return true;
+            }
+            """,
+            canEvaluate: nil,
+            isPrivate: false))
+
+        var exp = expectation(description: "will login")
+        var user: StitchUser!
+
+        try registerAndLogin { stitchUser in
+            user = stitchUser
+            exp.fulfill()
+        }
+
+        waitForExpectations(timeout: 10)
+
+        XCTAssertEqual(user.customData, [:])
+        exp = expectation(description: "will add favorite color")
+        client.callFunction(withName: "addUserProfile", withArgs: ["blue"], { (res: StitchResult<Bool>) in
+            switch res {
+            case .failure(let error):
+                XCTFail("could not add user profile: \(error.description)")
+            case .success(let result):
+                XCTAssert(result)
+            }
+            exp.fulfill()
+        })
+        waitForExpectations(timeout: 10)
+        XCTAssertEqual(user.customData, [:])
+
+        exp = expectation(description: "will refresh custom data")
+        client.auth.refreshCustomData({ _ in
+            exp.fulfill()
+        })
+        waitForExpectations(timeout: 10)
+
+        XCTAssertEqual(user.customData["favoriteColor"] as? String, "blue")
+
```

Comment:
[nit] since the `refreshCustomData` also refreshes an access token you might want to do a function call or something as this user to make sure the access token is updated